File: 409_longest_palindrome/longestPalindrome.ts
function longestPalindrome(s: string): number {
  const map = {};
  let res = 0;

  for (let i of s) {
    if (i in map) {
      delete map[i];
      res++;
    } else {
      map[i] = 1;
    }
  }

  if (!isEmpty(map)) {
    return res * 2 + 1;
  }

  return res * 2;
}

function isEmpty(obj) {
  for (const prop in obj) {
    if (Object.hasOwn(obj, prop)) {
      return false;
    }
  }

  return true;
}

File: 409_longest_palindrome/longestPalindrome.go
func longestPalindrome(s string) int {
	res := 0

	visited := make(map[rune]bool)

	for _, c := range s {
		if _, ok := visited[c]; ok {
			res += 2
			delete(visited, c)
		} else {
			visited[c] = true
		}
	}

	if len(visited) > 0 {
		res++
	}

	return res
}